compiler/src/java_plugin/cpp/java_generator.cpp

```diff
@@ -365,6 +365,21 @@ static void PrintMethodFields(
     (*vars)["arg_in_id"] = to_string(2 * i);
     (*vars)["arg_out_id"] = to_string(2 * i + 1);
     (*vars)["method_name"] = method->name();
+    bool safe = method->options().idempotency_level()
+        == google::protobuf::MethodOptions_IdempotencyLevel_NO_SIDE_EFFECTS;
```

Comment:
Wonder if we can add a safe unary method in golden test.proto.

Reply:
Works for me.  I can get to it tonight, or do a followup PR if this one can move forward.